src/service/PermissionService/index.ts

```diff
@@ -0,0 +1,462 @@
+import Keycloak from 'keycloak-js';
+
+import { getBearerTokenHeader } from '../../security/getBearerTokenHeader';
+import { getCsrfTokenHeader } from '../../security/getCsrfTokenHeader';
+
+import UserInstancePermission from '../../model/security/UserInstancePermission';
+import GroupInstancePermission from '../../model/security/GroupInstancePermission';
+import UserClassPermission from '../../model/security/UserClassPermission';
+import GroupClassPermission from '../../model/security/GroupClassPermission';
+import PermissionCollectionType from '../../model/enum/PermissionCollectionType';
+
+export interface PermissionServiceOpts {
+  basePath: string;
+  keycloak?: Keycloak;
```

Comment:
just a question and non blocking: shouldn't keycloak be required here?

Reply:
Yes, usually you'll always want to provide a `Keycloak` instance to the service, but generally speaking it's only needed if the backend interface requires a (Keycloak) bearer token for authentication. If the interface is accessible for anonymous users or no Keycloak is present at all, the client _might_ not be needed.

Reply:
allright, thanks for clarifying this :+1: